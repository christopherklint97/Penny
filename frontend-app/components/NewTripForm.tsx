import React from 'react'
import { useForm } from 'react-hook-form'

export default function NewTripForm() {
  const { register, handleSubmit, watch, errors } = useForm()
  const onSubmit = (data: any) => console.log(data)

  console.log(watch('search')) // watch input value by passing the name of it

  return (
    /* "handleSubmit" will validate your inputs before invoking "onSubmit" */
    <form className="text-left" onSubmit={handleSubmit(onSubmit)}>
      {/* register your input into the hook by invoking the "register" function */}
      <label htmlFor="cityName">Name of city</label>
      <input
        className="px-2 py-1 shadow-md focus:ring-2 focus:outline-none rounded-md mr-3 mb-2"
        name="cityName"
        ref={register({ required: true })}
        placeholder="London, Paris, Barcelona..."
      />
      {/* errors will return when field validation fails  */}
      {errors.search && (
        <p className="text-red-900 text-sm">This field is required</p>
      )}

      <input
        className="px-2 py-1 bg-green-200 rounded-md shadow-md hover:bg-green-300"
        type="submit"
        value="Add"
      />
    </form>
  )
}
