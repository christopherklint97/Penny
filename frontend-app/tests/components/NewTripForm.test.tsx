import React from 'react'
import { render } from '../testUtils'
import NewTripForm from '../../components/NewTripForm'

describe('NewTripForm component', () => {
  it('renders without crashing', function () {
    render(<NewTripForm onSubmit={jest.fn()} />)
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<NewTripForm onSubmit={jest.fn()} />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
