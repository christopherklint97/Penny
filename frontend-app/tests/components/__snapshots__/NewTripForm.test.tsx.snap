// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTripForm component matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="text-left"
  >
    <label
      class="block pb-1"
      for="city"
    >
      Name of city
    </label>
    <input
      class="px-2 py-1 shadow-md focus:ring-2 focus:outline-none rounded-md mr-3 mb-2 w-60"
      name="city"
      placeholder="London, Paris, Barcelona..."
    />
    <input
      class="px-2 py-1 bg-green-100 rounded-md shadow-md hover:bg-green-200 mb-4 cursor-pointer"
      type="submit"
      value="Search"
    />
  </form>
</DocumentFragment>
`;
